### PR TITLE
Add support for logging specific user out

### PR DESCRIPTION
**Pull request by cmelchior**
This PR adds support for logging individual users out instead of only the current user.

Note: Does not run the tests until Authentication tests are enabled.

### CODE REVIEW

File: src/sync/app.cpp [hunk 1]
```diff
@@ -586,10 +586,9 @@ void App::log_in_with_credentials(const AppCredentials& credentials,
     }, handler);
 }
 
-void App::log_out(std::function<void (Optional<AppError>)> completion_block) const
+void App::log_out(std::shared_ptr<SyncUser> user, std::function<void (Optional<AppError>)> completion_block) const
```
Comment (jsflax) [inline src/sync/app.cpp]: Should this be user_id instead? For easy compatibility.

Comment (cmelchior) [reply src/sync/app.cpp]: I don't have a strong opinion, either works for me. I'll change it.

Comment (cmelchior) [reply src/sync/app.cpp]: Or actually, no. 

There are some checks on the user inside `log_out`, so parsing the identity would require looking up the user again? That seems cumbersome and I don't think there are any cases where you don't have a reference to the shared pointer for the user for this?

Comment (jsflax) [reply src/sync/app.cpp]: Alright, that's fine.

**Pull request merged**